Sources/MetricKit/MetricsMonitor.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MetricsMonitor.swift
+//  
+//
+//  Created by Antoine van der Lee on 31/08/2021.
+//
+
+import Foundation
+import MetricKit
+
+/// Monitors payloads delivered by MetricKit and logs valueable information, including crashes.
+final class MetricsMonitor: NSObject {
+    
+    func startMonitoring() {
+        if #available(iOS 14, *) {
+            MXMetricManager.shared.add(self)
+        }
+        
+        NSSetUncaughtExceptionHandler { exception in
+            MetricsMonitor.logExceptionUsingCallStackSymbols(exception, description: "Uncaught Exception")
+        }
+    }
+    
+    /// Creates a new log section with the current thread call stack symbols.
+    private static func logExceptionUsingCallStackSymbols(_ exception: NSException, description: String) {
+        let message = """
+
+        ---
+
+        🚨 CRASH:
+        Description: \(description)
+        Exception name: \(exception.name.rawValue)
+        Reason: \(exception.reason ?? "nil")
+
+            \(Thread.callStackSymbols.joined(separator: "\n"))
+
+        ---
+
+        """
+        DiagnosticsLogger.standard.log(message)
+    }
+}
+
+extension MetricsMonitor: MXMetricManagerSubscriber {
+    @available(iOS 13.0, *)
+    func didReceive(_ payloads: [MXMetricPayload]) {
+        // We don't do anything with metrics yet.
+    }
+    
+    @available(iOS 14.0, *)
+    func didReceive(_ payloads: [MXDiagnosticPayload]) {
+        guard let payload = payloads.last else {
+            // We only use the last payload to prevent duplicate logging as much as possible.
+            return
+        }
+        
```

Comment:
<!--
  0 failure: 
  1 warning:  Lines should not ...
  
  
  DangerID: danger-id-Danger;
  File: Sources/MetricKit/MetricsMonitor.swift;
  Line: 56;
-->

- :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)